nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/AbstractHadoopProcessor.java

```diff
@@ -292,7 +304,11 @@ HdfsResources resetHDFSResources(String configResources, ProcessContext context)
                 ugi = SecurityUtil.loginKerberos(config, principal, keyTab);
                 fs = getFileSystemAsUser(config, ugi);
                 lastKerberosReloginTime = System.currentTimeMillis() / 1000;
-            } else {
+            } else if (context.getProperty(REMOTE_USER).isSet()){
```

Comment:
@bbende both points make perfect sense. Will make those changes. 

@trixpan - will look at adding the additional information you mentioned.